### PR TITLE
Add song category field

### PR DESCRIPTION
**Pull request by JFen9**
Category can only be one of "Special", "Christmas", or "Hymn". This is what Special means according to Steven.

> when we say special music, we mean something non-congregational, more performative… which we can play in a setting with a lot of nonXians to promote seeking Christianity, answering the big questions of life, like at a welcome night. I understand that some music is somewhere in between, or that other times we’ve retooled congregational songs to be more performative, but I think there aren’t too many outliers…

See https://vine.gracepointonline.org/t/topics-tags-in-gracetunes/12765 for more context.

Tested with
- create / update song page specifying category field
- song index page search with category filter
- show page with category field

https://github.com/PatrickF1/GraceTunes/assets/16906774/b7bc9bb1-ce8e-454f-a649-47fbff35c50d

### CODE REVIEW

**Comment by PatrickF1** [comment]
Thanks Joseph! Heads up we're blocked on Steven finalizing the list of tags. I don't have a ETA on that, unfortunately.

File: db/migrate/20240405155900_add_category_to_songs.rb
Comment (PatrickF1) [inline db/migrate/20240405155900_add_category_to_songs.rb:3]: Does this just default category to an empty string or NULL?

Comment (JFen9) [reply db/migrate/20240405155900_add_category_to_songs.rb:3]: yes, this will initialize the new column with all NULLs

File: app/models/song.rb
Comment (PatrickF1) [inline app/models/song.rb:18]: should we do some sort of validation on category?
```
validates :category, inclusion: { in: VALID_CATEGORIES, if: ->(song) { song.category.present? } }
```

**Comment by PatrickF1** [comment]
could you also add some tests please?

**Comment by PatrickF1** [comment]
> Heads up we're blocked on Steven finalizing the list of categories. I don't have a ETA on that, unfortunately.
Steven says he won't be able to figure out the categories until summer.

File: app/controllers/songs_controller.rb
Comment (PatrickF1) [inline app/controllers/songs_controller.rb:23]: We're hopefully eventually going to deprecate the server side rendering and use the API. Can you duplicate this change in the api/songs_controller please?

**Comment by PatrickF1** [comment]
Ok Steven is unresponsive. I'll drive this forward and find volunteers myself. I'll approve and merge this after you address my last comment.

File: app/models/song.rb
Comment (PatrickF1) [inline app/models/song.rb:15]: New request from Steven: let's just do 3 categories, remove easter

**Comment by PatrickF1** [comment]
Fixed all my feedback, then E2E tested. Thanks Joseph!

**Review by PatrickF1 — APPROVED** [review]

**Pull request merged**
